src/logic/calculate.js

```diff
@@ -1,32 +1,69 @@
 import { operate } from './operate';
 
-export const calculate = ({ total, next, operation }, btnName) => {
+export const calculate = ({ total, next, operation, displayed }, btnName) => {
   switch (btnName) {
     case 'AC':
+      displayed = false;
       total = '0';
       next = '0';
       operation = '';
       break;
 
-    case '+/-':
+    case '=':
+      displayed = true;
       total = operation ? operate(total, next, operation) : next;
       next = total;
-      operation = btnName;
+      operation = '';
+      if (total === "Error") total = "0";
       break;
 
-    case '%':
-      total = operation ? operate(total, next, operation) : next;
-      operation = btnName;
+    case '+/-':
+      displayed = true;
+      total = reverse(total);
+      next = total;
+      operation = '';
       break;
-
-    case '=':
+      
+    case "/":
+    case "x":
+    case "-":
```

Comment:
Let's use `'–'` here instead since both the Button name `prop` and `operate.js` is using `'–'`:

https://github.com/stiakov/math-magician-react-calc/blob/9275441e2bbac9faae5dfb5a85eff8ee4eec0018/src/components/ButtonPanel.js#L26 https://github.com/stiakov/math-magician-react-calc/blob/9275441e2bbac9faae5dfb5a85eff8ee4eec0018/src/logic/operate.js#L7

```suggestion
    case "–":
```

Alternatively, you can also decide to change the Button name `prop` and `operate.js` to `"-"`.

Currently, the operation is not recognising `'–'`: See screenshot below:

![2plus2](https://user-images.githubusercontent.com/36057474/70976424-3393a800-20ac-11ea-83b8-ff0a5f1e9d85.gif)